build-push/bin/main.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+# This script is not intended for humans.  It should be run by automation.
+# It's purpose is to produce a multi-arch container image based on the
+# contents of a prescribed git repository subdirectory structure.  The
+# URL for the git repository is the first command-line argument. The
+# second is the build context subdirectory, where the basename indicates
+# the image contents (upstream, testing, stable).
+
+set -eo pipefail
+
+if [[ -r "/etc/automation_environment" ]]; then
+    source /etc/automation_environment  # defines AUTOMATION_LIB_PATH
+    #shellcheck disable=SC1090,SC2154
+    source "$AUTOMATION_LIB_PATH/common_lib.sh"
+else
+    echo "Unexpected operating environment"
+    exit 1
+fi
+
+if [[ -z $(type -P build-push.sh) ]]; then
+    die "It does not appear that build-push.sh is installed properly"
+fi
+
+BPTMP=$(mktemp -p '' -d tmp-build-push-main-XXXX)
+
+# Assume transitive debugging state for build-push.sh if set
+export DEBUG
+
+# Arches to build by default
+ARCHES="${ARCHES:-amd64,ppc64le,s390x,arm64}"
+
+# First arg (REPO_URL) is the clone URL for repository ('main' branch only).
+REPO_URL="$1"
+REPO_NAME=$(basename "${REPO_URL%.git}")
+# Second arg (CTX_SUB) is the context subdirectory relative to the clone path
+CTX_SUB="$2"
+# Basename of second arg names the image contents
+CTX_NAME=$(basename "$CTX_SUB")
+_REG="quay.io"
+if [[ "$REPO_NAME" =~ testing ]]; then
+    _REG="example.com"
+fi
+REPO_FQIN="$_REG/$REPO_NAME/$CTX_NAME"
+req_env_vars REPO_URL REPO_NAME CTX_SUB CTX_NAME
+
+# shellcheck disable=SC2154
+dbg "$SCRIPT_FILENAME operating constants:
+    REPO_URL=$REPO_URL
+    REPO_NAME=$REPO_NAME
+    CTX_SUB=$CTX_SUB
+    CTX_NAME=$CTX_NAME
+    REPO_FQIN=$REPO_FQIN
+"
+
+# Set non-zero to avoid actually executing build-push, simply print
+# the command-line that would have been executed
+DRYRUN=${DRYRUN:0}
+_DRNOPUSH=""
+if ((DRYRUN)); then
+    _DRNOPUSH="--nopush"
+    warn "Operating in dry-run mode with $_DRNOPUSH"
+fi
+
+### MAIN
+
+cd "$BPTMP"
+dbg "Cloning $REPO_URL"
+git clone --no-local --no-hardlinks --depth 1 --single-branch --no-tags "$REPO_URL"
```

Comment:
Note to me: This script is most likely going to be run from CI with $PWD already in the subject repository.  Instead of cloning (again), simply verify `$PWD/.git` exists.  If it doesn't, then do the "clone + cd" operation.